internal/services/copilot_studio_application_insights/api_copilot_studio_application_insights.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package copilot_studio_application_insights
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/microsoft/terraform-provider-power-platform/internal/api"
+	"github.com/microsoft/terraform-provider-power-platform/internal/constants"
+)
+
+func newCopilotStudioClient(apiClient *api.Client) client {
+	return client{
+		Api: apiClient,
+	}
+}
+
+type client struct {
+	Api *api.Client
+}
+
+func (client *client) getCopilotStudioEndpoint(ctx context.Context, environmentId string) (string, error) {
+	env, err := client.getEnvironment(ctx, environmentId)
+	if err != nil {
+		return "", err
+	}
+	if env.Properties.RuntimeEndpoints.PowerVirtualAgents == "" {
+		return "", fmt.Errorf("Power Virtual Agents runtime endpoint is not available in the environment")
+	}
+
+	u, err := url.Parse(env.Properties.RuntimeEndpoints.PowerVirtualAgents)
+	if err != nil {
+		return "", err
+	}
+
+	return u.Host, nil
+}
+
+func (client *client) getEnvironment(ctx context.Context, environmentId string) (*EnvironmentIdDto, error) {
+	apiUrl := &url.URL{
+		Scheme: constants.HTTPS,
+		Host:   client.Api.GetConfig().Urls.BapiUrl,
+		Path:   fmt.Sprintf("/providers/Microsoft.BusinessAppPlatform/scopes/admin/environments/%s", environmentId),
+	}
+	values := url.Values{}
+	values.Add("api-version", "2023-06-01")
+	apiUrl.RawQuery = values.Encode()
+
+	env := EnvironmentIdDto{}
+	_, err := client.Api.Execute(ctx, nil, "GET", apiUrl.String(), nil, nil, []int{http.StatusOK}, &env)
+	if err != nil {
+		return nil, err
+	}
+
+	return &env, nil
+}
```

Comment:
@mawasile should we reuse existing environment functions instead of creating a new one?